system/jlib/jsocket.cpp

```diff
@@ -368,6 +368,73 @@ enum SOCKETMODE { sm_tcp_server, sm_tcp, sm_udp_server, sm_udp, sm_multicast_ser
 # endif
 #endif
 
+static CriticalSection queryKACS;
+
+enum UseKA { UNINIT, DISABLED, ENABLED };
+static std::atomic<UseKA> doKeepAlive { UNINIT };
+static int keepAliveTime = -1;
+static int keepAliveInterval = -1;
+static int keepAliveProbes = -1;
+
+/*
+<Software>
+  <Globals>
+    <expert>
+      <keepalive time="200" interval="75" probes="9"/>
+    </expert>
```

Comment:
in fact, \<expert\> is generated based on whatever is in Software/Globals.
It's because there are other sections generated into the bare-metal component configs from elsewhere (e.g. RemoteStorage), so these settings should look like:
```
Software>
  <Globals>
    <keepalive time="200" interval="75" probes="9"/>
  </Globals>
```

Reply:
updated